### PR TITLE
Feature: License list view

### LINKED ISSUE #66: Frontend licensee data list view
### Acceptance Criteria
- Fetch mock records from backend
- Sort by last name, home state, date updated
- Pagination (100 names per page)

### PR DESCRIPTION
**Pull request by jsandoval81**
### Description List
- Added the initial licensee list UI
    - Added the licensing data models & stores
    - Commented out some paging-related areas while the server API paging is in-progress
- Updated some yarn dependency minor versions

Closes #66

### CODE REVIEW

**Comment by waldoj** [comment]
Here's the review checklist that I'll work my way through in reviewing this PR:

- [ ] review the pull request to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/)
		- [ ] test in Mobile Safari and Mobile Chrome
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] review static code analysis results
- [ ] run a security audit of dependencies (e.g. `npm audit`) to ensure that there are no vulnerabilities that will be deployed to production (as opposed to vulnerabilities that only have an impact on the development environment)
- [ ] examine OWASP ZAP output to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, looking for problems (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] methodically review all new tests for correctness, quality of naming
	- [ ] look at code coverage of tests
	- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by waldoj** [comment]
@jsandoval81, is it a problem for your internal process if your pull requests back to CSG don't have the commits squashed? I'd prefer a merge commit, so that it's easy to review all of the commits that comprise a PR. It's a big help to start reviewing a PR by reading through all of the commits—when they're squashed, I need to hunt around in y'all's fork to find them.

**Comment by jusdino** [comment]
We tend to squash features to a single commit, but keep git history otherwise un-squashed. This keeps git history logically grouped into related changes, but suppresses the very noisy details of work in progress from making the history too difficult to browse. Is that not what you are seeing?

**Comment by waldoj** [comment]
I just see two commits, only one of substance ("Frontend license list UI," "Merge branch 'main' into development")...but also this PR is for just one feature, so the behavior is as you describe. If IA's preferred practice is to reduce Git commits to one per feature, it's not my place to ask you to change how you work.

**Review by waldoj — CHANGES_REQUESTED** [review]

**Comment by waldoj** [comment]
FYI, I'm yet to have `cdk destroy` execute correctly. For example:

```
# cdk destroy Sandbox/PersistentStack Sandbox/APIStack Sandbox/UIStack
Are you sure you want to delete: Sandbox/UIStack, Sandbox/APIStack, Sandbox/PersistentStack (y/n)? y
Sandbox/UIStack (Sandbox-UIStack): destroying... [1/3]

 ✅  Sandbox/UIStack (Sandbox-UIStack): destroyed
Sandbox/APIStack (Sandbox-APIStack): destroying... [2/3]

 ✅  Sandbox/APIStack (Sandbox-APIStack): destroyed
Sandbox/PersistentStack (Sandbox-PersistentStack): destroying... [3/3]
2:43:21 PM | DELETE_FAILED        | AWS::S3::Bucket                                   | AccessLogsBucket83982689
Resource handler returned message: "The bucket you tried to delete is not empty. You must delete all versions in the bucket. (Service: S3, Sta
tus Code: 409, Request ID: JA992YSHDFY8XM0R, Extended Request ID: 15ggzjSQQyTA0UuR5tow0ibOizsy7AzttNQsplRe2+z+s10sQb21iBMQhfShc/DoNiKX71jDSys=
)" (RequestToken: ffe83ba2-4c52-521c-d513-deda6139741a, HandlerErrorCode: GeneralServiceException)


 ❌  Sandbox/PersistentStack (Sandbox-PersistentStack): destroy failed Error: The stack named Sandbox-PersistentStack is in a failed state. You may need to delete it from the AWS console : DELETE_FAILED (The following resource(s) failed to delete: [AccessLogsBucket83982689]. ): Resource handler returned message: "The bucket you tried to delete is not empty. You must delete all versions in the bucket. (Service: S3, Status Code: 409, Request ID: JA992YSHDFY8XM0R, Extended Request ID: 15ggzjSQQyTA0UuR5tow0ibOizsy7AzttNQsplRe2+z+s10sQb21iBMQhfShc/DoNiKX71jDSys=)" (RequestToken: ffe83ba2-4c52-521c-d513-deda6139741a, HandlerErrorCode: GeneralServiceException)
    at destroyStack (/usr/local/lib/node_modules/aws-cdk/lib/index.js:455:2157)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async CdkToolkit.destroy (/usr/local/lib/node_modules/aws-cdk/lib/index.js:458:207603)
    at async exec4 (/usr/local/lib/node_modules/aws-cdk/lib/index.js:513:54331)

The stack named Sandbox-PersistentStack is in a failed state. You may need to delete it from the AWS console : DELETE_FAILED (The following resource(s) failed to delete: [AccessLogsBucket83982689]. ): Resource handler returned message: "The bucket you tried to delete is not empty. You must delete all versions in the bucket. (Service: S3, Status Code: 409, Request ID: JA992YSHDFY8XM0R, Extended Request ID: 15ggzjSQQyTA0UuR5tow0ibOizsy7AzttNQsplRe2+z+s10sQb21iBMQhfShc/DoNiKX71jDSys=)" (RequestToken: ffe83ba2-4c52-521c-d513-deda6139741a, HandlerErrorCode: GeneralServiceException)
```

I wind up having to delete the remaining artifacts manually. Because this isn't about deployment, but is about _destroying_ a deployment, this is not something that needs to be addressed for this pull request, but it is something that'll need to be addressed within a couple of sprints.

File: webroot/src/components/Lists/ListContainer/ListContainer.spec.ts [hunk 1]
```diff
@@ -58,9 +56,8 @@ describe('ListContainer component', async () => {
         });
 
         expect(wrapper.find('.no-records').exists(), 'no records').to.equal(false);
-        expect(wrapper.findAllComponents(Sorting).length, 'sorting elements').to.equal(1);
+        expect(wrapper.findAllComponents(Sorting).length, 'sorting elements').to.equal(0);
```
[the remark this answers is not in the document]
Comment (jsandoval81) [reply webroot/src/components/Lists/ListContainer/ListContainer.spec.ts:59]: Good question. The default framework from the first sprint had list sorting as a separate UI component above the list. The latest UI designs call for sorting to be handled directly on the column headers. So the previous, separate `Sorting` component was removed from this parent component.

File: webroot/src/store/user/user.actions.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ export default {
         dispatch('clearAuthTokens');
         commit(MutationTypes.LOGOUT_REQUEST);
 
-        /* istanbul ignore if */
```
[the remark this answers is not in the document]
Comment (jsandoval81) [reply webroot/src/store/user/user.actions.ts:38]: Good question. We aim to use this _really_ sparingly, such as cases where the code is only reached in local development contexts _and_ complex or impractical to create mocks. It does get skipped by coverage instrumentation. I don't think the 1 or 2 uses currently has a material impact on coverage. 

And we're totally fine (& expecting) each of these to be questioned as they're introduced! 👍 They should only represent far-edge cases, not just random usage.

[the remark this answers is not in the document]
Comment (waldoj) [reply webroot/src/store/user/user.actions.ts:38]: Sounds good—thank you.

File: webroot/src/components/Lists/ListContainer/ListContainer.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (waldoj) [reply webroot/src/components/Lists/ListContainer/ListContainer.spec.ts:59]: Ah-ha, that makes sense—thank you.

**Review by waldoj — APPROVED** [review]

**Comment by waldoj** [comment]
@isabeleliassen, I have reviewed this pull request, evaluating it for conformance with the QASP, and conducting a code review. I recommend that the pull request be merged into CSG's repo.

**Pull request merged**